### PR TITLE
Start testing on Django v5.1

### PR DESCRIPTION
**Pull request by cclauss**
Test the **pre-release** version of Django v5.1 scheduled to be released in August 2024.
* https://docs.djangoproject.com/en/dev/releases/5.1

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -14,17 +14,19 @@ jobs:
...
         python-version: ['3.8', '3.12']
-        django-version: [3.2, 4.2, 5.0]
+        django-version: [3.2, 4.2, 5.0, 5.1a1]
         exclude:
           - python-version: "3.12"
             django-version: "3.2"
           - python-version: "3.8"
             django-version: "5.0"
+          - python-version: "3.8"
+            django-version: "5.1a1"
```
Comment (cclauss) [inline .github/workflows/test.yml:24]:
```suggestion
        django-version: [3.2, 4.2, 5.0, 5.1b1]
        exclude:
          - python-version: "3.12"
            django-version: "3.2"
          - python-version: "3.8"
            django-version: "5.0"
          - python-version: "3.8"
            django-version: "5.1b1"
```

**Comment by jamesturk** [comment]
thanks!

**Comment by cclauss** [comment]
Why does upgrading Django v5.0 --> v5.1b1 ___downgrade___`django-honeypot`?
```
django-honeypot: 1.2.0 --> 0.9.0

2,3c2,4
< Django==5.0.6
< django-honeypot==1.2.0
---
> Django==5.1b1
> django-honeypot==0.9.0
> setuptools==70.1.1
```
---
```
mkdir django-honeypot-on-dj50 \
 && cd django-honeypot-on-dj50 \
 && python3.12 -m venv venv \
 && source venv/bin/activate \
 && pip install --upgrade pip \
 && pip install Django==5.0.6 django-honeypot \
 && pip freeze > requirements.txt \
 && cd .. \
 && mkdir django-honeypot-on-dj51b1 \
 && cd django-honeypot-on-dj51b1 \
 && python3.12 -m venv venv \
 && source venv/bin/activate \
 && pip install --upgrade pip \
 && pip install Django==5.1b1 django-honeypot \
 && pip freeze > requirements.txt \
 && cd .. \
 && diff django-honeypot-on-dj50/requirements.txt  \
         django-honeypot-on-dj51b1/requirements.txt \
 && rm -fr django-honeypot-on-dj5*
 ```

**Comment by cclauss** [comment]
@jamesturk Can we please get a release to PyPI so the downgrade behavior discussed above stops?

@Mogost Your thoughts?

**Comment by jamesturk** [comment]
pushed a 1.2.1 that should have the right classifiers, lmk if there are issues

**Pull request merged**